### PR TITLE
SAK-46023 Date Manager > Nullable fields for some activities needs to be allowed

### PR DESCRIPTION
**Pull request by gwickram**
https://sakaiproject.atlassian.net/browse/SAK-46023

### CODE REVIEW

File: site-manage/datemanager/impl/src/java/org/sakaiproject/datemanager/impl/DateManagerServiceImpl.java [hunk 1]
```diff
@@ -437,11 +454,26 @@ public DateManagerValidation validateAssessments(String siteId, JSONArray assess
...
-					errored = errors.add(new DateManagerError("due_date", rb.getString("error.due.date.not.found"), "assessments", toolTitle, idx));
-				}
-				if (acceptUntil == null && lateHandling) {
-					errored = errors.add(new DateManagerError("accept_until", rb.getString("error.accept.until.not.found"), "assessments", toolTitle, idx));
+				if (acceptUntil != null) {
+					if (dueDate == null) {
+						errors.add(
+								new DateManagerError(
+										"accept_until",
+										"You need to set a valid Due Date if the assessment has a Late Acceptance Date.",
```
Comment (jesusmmp) [inline site-manage/datemanager/impl/src/java/org/sakaiproject/datemanager/impl/DateManagerServiceImpl.java:462]: Please use rb.getString with a property for i18n

File: site-manage/datemanager/impl/src/java/org/sakaiproject/datemanager/impl/DateManagerServiceImpl.java [hunk 2]
```diff
@@ -437,11 +454,26 @@ public DateManagerValidation validateAssessments(String siteId, JSONArray assess
...
+										"assessments",
+										toolTitle,
+										idx));
+						errored = true;
+					} else if (acceptUntil.isAfter(dueDate)) {
+						errors.add(
+								new DateManagerError(
+										"accept_until",
+										"The Late Submission Date cannot be earlier than the Available Date or the Due Date.",
```
Comment (jesusmmp) [inline site-manage/datemanager/impl/src/java/org/sakaiproject/datemanager/impl/DateManagerServiceImpl.java:471]: Please use rb.getString with a property for i18n

File: site-manage/datemanager/impl/src/java/org/sakaiproject/datemanager/impl/DateManagerServiceImpl.java (hunk 1 above)
Comment (gwickram) [reply site-manage/datemanager/impl/src/java/org/sakaiproject/datemanager/impl/DateManagerServiceImpl.java]: Moved the error messages to properties file.

File: site-manage/datemanager/impl/src/java/org/sakaiproject/datemanager/impl/DateManagerServiceImpl.java (hunk 2 above)
Comment (gwickram) [reply site-manage/datemanager/impl/src/java/org/sakaiproject/datemanager/impl/DateManagerServiceImpl.java]: Moved the error messages to properties file.

**Review by ottenhoff — APPROVED** [review]
We made a few edits while on the core call to improve formatting.

Let's wait until this passes the build tests and then merge it in!

File: site-manage/datemanager/impl/src/java/org/sakaiproject/datemanager/impl/DateManagerServiceImpl.java [hunk 3]
```diff
@@ -512,11 +532,12 @@ public void updateAssessments(DateManagerValidation assessmentsValidation) throw
...
-				if (update.dueDate != null) {
-					control.setDueDate(Date.from(update.dueDate));
-				}
-				if (lateHandling && update.acceptUntilDate != null) {
-					control.setRetractDate(Date.from(update.acceptUntilDate));
+				Date dueDateTemp = update.dueDate != null ? Date.from(update.dueDate) : null;
+				control.setDueDate(dueDateTemp);
+				if (lateHandling) {
+					Date lateDateTemp =
+							update.acceptUntilDate != null ? Date.from(update.acceptUntilDate) : null;
```
Comment (ottenhoff) [inline site-manage/datemanager/impl/src/java/org/sakaiproject/datemanager/impl/DateManagerServiceImpl.java:539]: would prefer these two lines to be on one line per the conventions above

File: site-manage/datemanager/impl/src/java/org/sakaiproject/datemanager/impl/DateManagerServiceImpl.java [hunk 4]
```diff
@@ -603,20 +625,21 @@ public DateManagerValidation validateGradebookItems(String siteId, JSONArray gra
...
-
-				if (dueDate == null) {
-					errors.add(new DateManagerError("due_date", rb.getString("error.due.date.not.found"), "gradebookItems", toolTitle, idx));
-					continue;
-				}
+				String dueDateRaw = (String) jsonItem.get("due_date");
+				Instant dueDate = null;
+				if (dueDateRaw != null && !dueDateRaw.isEmpty())
+					dueDate =
+							userTimeService
```
Comment (ottenhoff) [inline site-manage/datemanager/impl/src/java/org/sakaiproject/datemanager/impl/DateManagerServiceImpl.java:632]: I would prefer less line breaks here.

**Pull request merged**